### DISCUSSION TITLE
Production readiness of AvaloniaUI

### DISCUSSION THREAD

**Discussion by llfab** [AvaloniaUI/Avalonia#6321] [General]
Dear all,

I would like to get your insights about the following controversy:
- Microsoft now targets more and more cross-platform development and what they do with .net 5 and 6 is great. However, they provide no solution for cross-platform UI development. At least with WinUI 3 they have gone a big step forward for the Windows OS. .Net MAUI does not really target desktop application development which is underlined by the fact that MAUI connects to the native OS controls and that Linux is not in MAUI's focus
- UNO platform as another solution seems not to target full fletched desktop applications having access to all local resources. 
- Hence, AvaloniaUI really seems like the missing bit of technology. Having used Qt/QML (which works on Linux, mac and Windows in production ready fashion) for medical device applications I am sure that such cross-platform technologies do work if they are done well and are well maintained
- I now read yet another article from Holger Schwichtenberg (https://www.heise.de/news/Kommentar-So-kann-Microsoft-die-Abwanderung-von-NET-Entwicklern-nicht-stoppen-4725901.html). He states that Microsoft fails in delivering a true cross-platform UI framework but at the same time warns the community to make your product dependent on an "unreliable" provider like Avalonia. 
- I ask myself a similar question. I love what I see about AvaloniaUI but I am also a bit hesitant of (first) adding yet another dependency to a medical device UI project and (second) to make us dependent on a project that may be shut down like it happened with so many other promising projects (e.g., SharpDX)

So, how do you guys think about this? 
What would be your response to the "unreliable source" comment in Schwichtenberg's blog?

**Comment by mxsoftware** [comment]
I might be a little off topic and I apologize in advance for my English.
Over the years I have been developing desktop applications for Windows and seeing them made by third parties.
Remaining in the WPF world I don't remember a desktop application (including mine) that didn't use MahApps.Metro, Prism or ReactiveUI.
If I'm not mistaken these are also opensource products that are widely used in commercial software. If you do a statistic, the messages generated by Github for these products are tens of times lower than those for Avalonia.
In summary: we have many desktop products including commercial ones that using open source libraries because obviously this provided by Microsoft is not enough, Avalonia UI has a decidedly higher volume of messages generated weekly when compared to all these projects put together, what conclusion can we draw?

**Comment by mxsoftware** [comment]
One more question (which answers itself): what have been the technological improvements and adaptations included in Wpf from 2006 to today?
And are we really sure that the new platforms proposed by Microsoft for the development of desktop applications meet all the needs of developers?

**Reply by llfab** [reply]
I agree, the improvements in WPF have been close to zero. The fact that it is still on DX9 basically tells the whole story. Therefore, I believe winui 3 is the long awaited right step. However, for Windows only.

**Reply by llfab** [reply]
In addition: At least from what I am seeing WinUI 3 seems a good improvement (finally). It will provide latest DX support, compile time bindings and SwapChainPanel to integrate your own renderer engine. So for Windows this seems very much ok. But you are right. For (desktop) cross-platform (Win, Linux, macOS) there is nothing on the horizon, which the community criticizes a lot. I also believe that this will continue to cost Microsoft more and more.

**Comment by robloo** [comment]
At the link is high level comparison of cross platform XAML frameworks. You might find it relevant to this discussion.

https://github.com/robloo/PublicDocs/blob/master/XAMLFrameworkComparison.md#comparison-table

Note that several companies are now depending on Avalonia so it certainly is ready for production. Paid support is also an option for companies that want to ensure their needs are always met.
